src/GitVersionCore/ExecuteCore.cs

```diff
@@ -57,7 +58,14 @@ public VersionVariables ExecuteGitVersion(string targetUrl, string dynamicReposi
                 if (versionVariables == null)
                 {
                     versionVariables = ExecuteInternal(targetBranch, commitId, repo, gitPreparer, projectRoot, buildServer, overrideConfig: overrideConfig);
-                    gitVersionCache.WriteVariablesToDiskCache(repo, dotGitDirectory, versionVariables);
+                    try
+                    {
+                        gitVersionCache.WriteVariablesToDiskCache(repo, dotGitDirectory, versionVariables);
+                    }
+                    catch (IOException)
+                    {
+                        Logger.WriteInfo("I/O exception during cache write");
```

Comment:
I suspect that a warning result in an msbuild warning being raised? If that is the case I would be careful about using it for this behavior since some CI build systems fail the build on warnings and this is not something that should impact the build in my opinion. But I could be wrong about the msbuild warning.